doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -1,175 +1,498 @@
+    {
+      std::string actionName = name + "/" + action_ns;
+
+      actionClient_ =
+          std::make_shared<actionlib::SimpleActionClient<control_msgs::FollowJointTrajectoryAction>>(actionName, true);
+
+      // Timeout can be loaded from settings
+      actionClient_->waitForServer(ros::Duration(20.0));
+
+      if (!actionClient_->isServerConnected())
+      {
+        // Report connection error
+        actionClient_.reset();
+      }
+    }
+
+  public:
+    // MoveIt calls this method when it wants to send a trajectory goal to execute
+    bool sendTrajectory(const moveit_msgs::RobotTrajectory& trajectory) override
+    {
+      if (!actionClient_)
+      {
+        // Report connection error
+        return false;
+      }
+
+      control_msgs::FollowJointTrajectoryGoal goal;
+      goal.trajectory = trajectory.joint_trajectory;
+
+      actionClient_->sendGoal(
+          goal,
+          [this](const auto& state, const auto& result) {
+            // Complete trajectory callback
+            done_ = true;
+          },
+          [this] {
+            // Begin trajectory callback
+          },
+          [this](const auto& feedback) {
+            // Trajectory state callback
+          });
+
+      done_ = false;
+
+      return true;
+    }
+
+    // MoveIt calls this method when it wants a blocking call until done
+    bool waitForExecution(const ros::Duration& timeout = ros::Duration(0)) override
+    {
+      if (actionClient_ && !done_)
+        return actionClient_->waitForResult(ros::Duration(5.0));
+
+      return true;
+    }
+
+    // MoveIt calls this method to get status updates
+    moveit_controller_manager::ExecutionStatus getLastExecutionStatus() override
+    {
+      // Report last status here
+      return moveit_controller_manager::ExecutionStatus::SUCCEEDED;
+    }
+
+    // MoveIt calls this method to abort trajectory goal execution
+    bool cancelExecution() override
+    {
+      if (!actionClient_)
+        return false;
+
+      actionClient_->cancelGoal();
+      done_ = true;
+
+      return true;
+    }
+  };
+  }  // namespace example
+
+.. note::
+   Replace ``your_controller_action`` with the type of action interface supported by the controller, and ``your_timeout`` with how long to wait for the connection to take place (this can be read from settings). If the controller doesn't support an Action Server, this can be replaced by whichever mechanism is supported.
+
+Once implemented, the controller handle does not need to be exported, since it's returned by the controller handle allocator, which is exported.
+
+Custom Controllers and Managers
+-------------------------------
+
+Custom controllers not managed by *ROS Controller Manager* can be integrated by implementing MoveIt *Controller Handle* and *Controller Handle Allocator* that conform to the MoveIt interface. Then they can be loaded by *MoveIt Simple Controller Manager* as described earlier in this topic.
```

Comment:
This is wrong. MoveItSimpleControllerManager just supports the FollowJointTrajectory or GripperCommand action, no matter the underlying controller is a ros_control controller or not. There is no way to expand that interface. The ControllerHandleAllocator plugin is tied to the ROSControllerManager.

Reply:
I will take a look at re-writing that

Reply:
Take a look at the new text I wrote.

Reply:
I like that you re-organized it by MoveIt controller manager type. I thought about doing that but wasn't sure if it could be done while maintaining a clear narrative. The way you re-wrote it here looks great.